pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -447,12 +449,40 @@ func waitForOpenStackLb(client *gophercloud.ServiceClient, lbId string) error {
 
 // Looks for a Octavia load balancer by name, address and subnet ID. If it does
 // not exist creates it. Will fail if multiple LB's are matching all criteria.
-func ensureOpenStackLb(client *gophercloud.ServiceClient, name, vipAddress, vipSubnetId string) (string, error) {
-	page, err := loadbalancers.List(client, loadbalancers.ListOpts{
+func ensureOpenStackLb(client *gophercloud.ServiceClient, name, vipAddress, vipSubnetId, tag string) (string, error) {
+	// We need to figure out if Octavia supports tags and use description field if it's too old. To do that
+	// we list available API versions and look for 2.5.
+	allPages, err := apiversions.List(client).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	apiVersions, err := apiversions.ExtractAPIVersions(allPages)
+	if err != nil {
+		return "", err
+	}
+
+	var octaviaTagSupport bool
+	octaviaTagSupport = false
+	for _, apiVersion := range apiVersions {
+		if apiVersion.ID >= MinOctaviaVersionWithTagSupport {
```

Comment:
I think you need to use a real semver parser here...

Reply:
Yeah, this probably deserves a comment, it took me a while to understand @MaysaMacedo code, on which I based this.

Basically Octavia returns all the API versions it supports, so once we hit any of v2.5, v2.6, v2.7, v2.8, v2.9, we're assuming everything's fine.

Reply:
Okay, so we discovered we'll be doing more Octavia version detection, so I use proper semver parser now.